tests/python/test_with_sklearn.py

```diff
@@ -1526,6 +1527,58 @@ def test_tags() -> None:
     assert "multioutput" not in tags
 
 
+# the try-excepts in this test should be removed once xgboost's
```

Comment:
We can use `pytest.mark.skipif` to skip tests. Seems simpler.

Reply:
I was thinking that it's useful to check that the *exact, expected* `AttributeError` is raised...there are so many layers of inheritance involved here, it wouldn't be hard to implement this in a way that accidentally raises some totally unrelated error accessing a non-existent property or something.

If you read that and still think `skipif()` would be preferable, I'll happy change to that and remove the `try-except`, just wanted to explain my thinking.